framework/web/UrlManager.php

```diff
@@ -428,6 +429,27 @@ public function createUrl($params)
     }
 
     /**
+     * Returns the value indicating whether result of [[createUrl()]] of rule should be cached in internal cache.
+     *
+     * @param UrlRuleInterface $rule
+     * @return bool|null `true` if result should be cached, `false` if not, `null` if rule does not provide info about
+     * create URL status.
+     * @since 2.0.12
+     * @see getUrlFromCache()
+     * @see setRuleToCache()
+     * @see UrlRule::$createStatus
+     */
+    protected function canBeCached(UrlRuleInterface $rule)
```

Comment:
That was my first shot, but I'm not sure if rule should decide about UrlManager internal cache logic. I'm open to suggestions in this case.